kzg/src/eth/eip_7594.rs

```diff
@@ -0,0 +1,146 @@
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+use alloc::{format, string::String};
+
+use crate::{
+    cfg_iter,
+    das::{EcBackend, DAS},
+    eip_4844::bytes_to_blob,
+    eth::{
+        BYTES_PER_BLOB, BYTES_PER_CELL, BYTES_PER_COMMITMENT, BYTES_PER_FIELD_ELEMENT,
+        BYTES_PER_PROOF, CELLS_PER_EXT_BLOB, FIELD_ELEMENTS_PER_CELL, FIELD_ELEMENTS_PER_EXT_BLOB,
+    },
+    Fr, G1,
+};
+
+pub type CellsKzgProofs = (Vec<[u8; BYTES_PER_CELL]>, Vec<[u8; BYTES_PER_PROOF]>);
+
+pub fn recover_cells_and_kzg_proofs_raw<B: EcBackend>(
+    cell_indices: &[usize],
+    cells: &[[u8; BYTES_PER_CELL]],
+    das: &impl DAS<B>,
+) -> Result<CellsKzgProofs, String>
+where
+    B::G1: Copy,
+    B::Fr: Copy,
+{
+    let nested_cells = cfg_iter!(cells)
+        .map(|bytes| bytes_to_cell::<B>(bytes))
+        .collect::<Result<Vec<Vec<_>>, _>>()?;
+    let cells = nested_cells.into_iter().flatten().collect::<Vec<_>>();
+
+    let mut recovered_cells = [B::Fr::default(); FIELD_ELEMENTS_PER_EXT_BLOB];
+    let mut recovered_proofs = [B::G1::default(); CELLS_PER_EXT_BLOB];
+
+    let _ = das.recover_cells_and_kzg_proofs(
+        &mut recovered_cells,
+        Some(&mut recovered_proofs),
+        cell_indices,
+        &cells,
+    );
```

Comment:
I believe there should be `?`, because you're currently ignoring error, if it occurs in recover_cells_and_kzg_proofs:
```suggestion
    das.recover_cells_and_kzg_proofs(
        &mut recovered_cells,
        Some(&mut recovered_proofs),
        cell_indices,
        &cells,
    )?;
```

Reply:
ahh! good eyes